File: python/check_compatibility.py
from client import create_native_client
import nativeconnect.api.v1.nativeconnect_pb2 as native


def main():
    client = create_native_client()

    request = native.CheckCompatibilityRequest(
        vin="LRW3F7FS9PC893577",
        region=native.Region.COUNTRY_AUS,
        usecase=native.Usecase.USECASE_B2C
    )

    response = client.CheckCompatibility(request)

    print(response)


if __name__ == "__main__":
    main()

File: typescript/src/check_compatibility.ts
import { createNodeClient } from "./client"
import { create, toJsonString } from "@bufbuild/protobuf";
import {
  CheckCompatibilityRequestSchema,
  CheckCompatibilityResponseSchema,
  Region,
  Usecase
} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb";

const client = createNodeClient()

const request = create(CheckCompatibilityRequestSchema, {
  vin: "LRW3F7FS9PC893577",
  region: Region.COUNTRY_AUS,
  usecase: Usecase.B2C
})

const response = await client.checkCompatibility(request)

console.log(toJsonString(CheckCompatibilityResponseSchema, response, { prettySpaces: 2 }))
